apps/snitch_core/lib/core/data/model/promotion/promotion_applicability.ex

```diff
@@ -0,0 +1,91 @@
+defmodule Snitch.Data.Model.Promotion.Applicability do
+  @moduledoc """
+  Exposes functions related to
+  """
+
+  @doc """
+  Checks if a coupon with the supplied code exists.
+  """
+  use Snitch.Data.Model
+  alias Snitch.Data.Schema.Promotion
+
+  @errors %{
+    not_found: "promotion not found",
+    inactive: "promotion is not active",
+    expired: "promotion has expired"
+  }
+
+  @spec valid_coupon_check(String.t()) :: {:ok, Promotion.t()} | {:error, String.t()}
+  def valid_coupon_check(coupon_code) do
+    case Repo.get_by(Promotion, code: coupon_code) do
+      nil ->
+        {:error, @errors.not_found}
+
+      promotion ->
+        {:ok, promotion}
+    end
+  end
+
+  def promotion_active?(promotion) do
```

Comment:
modify function name.

Response format should be consistent for all the functions.